### PR TITLE
#728 Encrypt user_id on account verification

### PR DESCRIPTION
**Pull request by OmarElgabry**
Instead of passing user_id in the URL as action method argument, passing the encrypted version of user_id in query parameter will work just fine. The link looks long and ugly :smile:  but in case you want a solution for #728

### CODE REVIEW

**Comment by panique** [comment]
Thanks, looks very good!
Would be cool if everybody who reads this could test it a little bit to make sure this is bulletproof in most possible browser/mailprovider/server/os-setup (there are sometimes problems when using complex strings with special characters in URLs).

File: application/controller/RegisterController.php [hunk 1]
```diff
@@ -46,11 +46,12 @@ public function register_action()
 
     /**
      * Verify user after activation mail link opened
-     * @param int $user_id user's id
      * @param string $user_activation_verification_code user's verification token
      */
-    public function verify($user_id, $user_activation_verification_code)
+    public function verify($user_activation_verification_code)
     {
+		$user_id = Encryption::decrypt(Request::get("user_id"));
```
Comment (slaveek) [inline application/controller/RegisterController.php]: Request::get("user_id") should be check for empty value otherwise exception is thrown

**Comment by OmarElGabry** [comment]
@slaveek It's fixed now, Thanks.

**Comment by borgogelli** [comment]
You change the sendVerificationEmail() but what about sendPasswordResetMail() ? Do I need to change it too ?

**Comment by OmarElGabry** [comment]
The password rest uses the user name instead of user id, that's why I changed sendVerificationEmail(). You can use the user id instead in sendPasswordResetMail(), encrypt it, and don't forget to pass it as a query argument.

**Comment by di48l069** [comment]
Hi @panique Will you be adding this to huge?

**Comment by panique** [comment]
Hi @di48l069, yes it's the plan but i had no time for review and merging yet :/ ... you know, working full time + summer is always tricky :)

**Comment by di48l069** [comment]
Hi @panique thank you for response haha I understand working full time is hectic no problem man i will just keep eye on pull request :P

Thanks for this amazing secure framework it has helped my projects alot

**Comment by Kayumba** [comment]
v

**Pull request closed without merging**